File: paw/interfaces/src/main/java/ar/edu/itba/paw/interfaces/services/LocationService.java
package ar.edu.itba.paw.interfaces.services;

import ar.edu.itba.paw.model.location.City;
import ar.edu.itba.paw.model.location.Country;
import ar.edu.itba.paw.model.location.State;

import java.util.List;

public interface LocationService {

    /**
     * Gets the list of all countries.
     * @return The list of all the countries.
     */
    List<Country> findAllCountries();


    /**
     * Gets all the states for a given country.
     * @param countryId The unique id of the country.
     * @return The list of the country's respective states.
     */
    List<State> findStates(long countryId);


    /**
     * Gets all the cities for a given state.
     * @param stateId The unique id of the state.
     * @return The list of the state's respective cities.
     */
    List<City> findCities(long stateId);
}

File: paw/angular/app/scripts/services/locationService.js
define(['paw2020a', 'services/AuthenticatedRestangular'], function(paw2020a) {

  'use strict';
  paw2020a.service('locationService', ['AuthenticatedRestangular', function(AuthenticatedRestangular) {
    var locationService = {};

    var root = AuthenticatedRestangular.all('location');

    locationService.getCountryList = function () {
      return root.all('country').getList();
    };


    locationService.getStateList = function (countryId) {
      return root.all('state').all(countryId).getList();
    };


    locationService.getCityList = function (stateId) {
      return root.all('city').all(stateId).getList();
    };


    return locationService;
  }]);

});
